arviz/stats/stats.py

```diff
@@ -430,29 +447,60 @@ def loo(data, pointwise=False, reff=None, scale="deviance"):
         )
         warn_mg = True
 
-    loo_lppd_i = scale_value * _logsumexp(log_weights, axis=0)
-    loo_lppd = loo_lppd_i.sum()
-    loo_lppd_se = (len(loo_lppd_i) * np.var(loo_lppd_i)) ** 0.5
-
-    lppd = np.sum(_logsumexp(log_likelihood, axis=0, b_inv=log_likelihood.shape[0]))
+    ufunc_kwargs = {"n_dims": 1, "ravel": False}
+    kwargs = {"input_core_dims": [["samples"]]}
+    loo_lppd_i = scale_value * _wrap_xarray_ufunc(
+        _logsumexp, log_weights, ufunc_kwargs=ufunc_kwargs, **kwargs
+    )
+    loo_lppd = loo_lppd_i.values.sum()
+    loo_lppd_se = (n_data_points * np.var(loo_lppd_i.values)) ** 0.5
+
+    lppd = np.sum(
+        _wrap_xarray_ufunc(
+            _logsumexp,
+            log_likelihood,
+            func_kwargs={"b_inv": n_samples},
+            ufunc_kwargs=ufunc_kwargs,
+            **kwargs
+        ).values
+    )
```

Comment:
None. The computations performed are the same, I have not modified how logsumexp works. I did this so that there is no need to convert to arrays and the named dims and coords info is not lost.